Use Jetbrains httpsupport, and remove lots of unused imports/aspects

malueck opened the pull request:
Removed the URLUtil class (and jetty.io dependency) and use jetbrains.mps.ide.httpsupport instead, which does exactly the same but is part of MPS now.

Also removed lots of unused imports and empty aspects.

In the build script, added as plugin dependencies:
de.slisson.mps.conditionalEditor
de.itemis.mps.celllayout
de.itemis.mps.grammarcells
com.mbeddr.mpsutil.intentions

For this, I am not sure if it is the correct approach to add these like this for all used languages / runtime dependencies. We would that the plugin will disable itself in the MPS Plugin settings and say for example "com.mbeddr.mpsutil.intentions is missing" if this is the case, and at the moment it simply throws an exception on start if com.mbeddr.mpsutil.intentions is absent. Also, we would like that the model checker would find an error in the build script if a dependency is not satisfied. However, I am not sure if it already doesn't do so and it is just the job of the plugin to declare its dependencies somehow differently (adding those as "Dependencies" to the language will show them as unused).
Any thoughts on this? Is this step with adding build dependencies the correct way?

malueck commented:
This will resolve item 2) of https://github.com/IETS3/iets3.opensource/issues/1207: org.iets3.core.expr.stringvalidation.runtime has dependency on iets3.core.assessment that is not in the build script.

alexanderpann commented:
Your change with the httpsupport plugin basically reverts https://github.com/IETS3/iets3.opensource/commit/2ebc1181f55b9180b2b159d0df88940a11bfa2e8. To be on the safe side, we should probably add the httpsupport plugin [here](http://127.0.0.1:63320/node?ref=r%3Ac3d6ae0c-8b10-477f-a3e9-5dc8700ceb13%28org.iets3.opensource.build.build%29%2F1893833104133534633) as an additional ideal plugin. Other than that, I am fine with the changes.

malueck commented:
Oops, you are right, the httpsupport use was indeed removed on purpose. But if this is fine -- I added it as loaded plugin to the tests as you suggested. The tests seem to pass, is there anything else to check that the issue with httpsupport does not appear again?
Looking at the other commit message, I am not sure what "tests executed out of process" means in this context. When I close MPS and run them with gradlew, they pass, but perhaps there is more to that?

I also added com.mbeddr.mpsutil.userstyles as a build dependency, which is a used language through org.iets3.core.base, similar to com.mbeddr.mpsutil.intentions. Like the other one, userstyles was not yet a build dependency (neither direct nor through mbeddr.core.base) but having it missing makes all the IETS3 languages not deploy because they use it. The build solution should now take care of that.

alexanderpann reviewed (approved):
"tests executed out of process" referred to the test run configurations where you can make the tests execute in-process or out-of-process. They don't work properly anyway in some cases, so let's just merge this PR to fix the other issues.
